frankenphp.go

```diff
@@ -444,19 +444,23 @@ func go_register_variables(rh C.uintptr_t, trackVarsArray *C.zval) {
 	r := cgo.Handle(rh).Value().(*http.Request)
 	env = r.Context().Value(contextKey).(*FrankenPHPContext).Env
 
-	// FIXME: remove this debug statement
-	env[fmt.Sprintf("REQUEST_%d", rh)] = "1"
+	le := len(env) * 2
+	cArr := (**C.char)(C.malloc(C.size_t(le) * C.size_t(unsafe.Sizeof((*C.char)(nil)))))
+	variables := unsafe.Slice(cArr, le)
 
-	// TODO: batch this for performance
+	var i int
 	for k, v := range env {
-		ck := C.CString(k)
-		cv := C.CString(v)
-
-		C.php_register_variable(ck, cv, trackVarsArray)
-
-		C.free(unsafe.Pointer(ck))
-		C.free(unsafe.Pointer(cv))
+		variables[i] = C.CString(k)
```

Comment:
Maybe could we allocate all the memory we need in one batch (to have contiguous memory)? Not sure if it is worth it